enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/OtpServiceImpl.java

```diff
@@ -131,17 +131,17 @@ public String createOtpCodeForResend(OnboardingProcessEntity process, OtpType ot
      */
     public void cancelOtp(OnboardingProcessEntity process, OtpType otpType) {
         final String processId = process.getId();
-        // Fail current OTP, if it is present
+        // Fail current OTP, if it is present and in ACTIVE state
         onboardingOtpRepository.findNewestByProcessIdAndType(processId, otpType).ifPresent(otp -> {
             final Date now = new Date();
-            if (otp.getStatus() != OtpStatus.FAILED) {
+            if (otp.getStatus() == OtpStatus.ACTIVE) {
```

Comment:
Ok, I have double-checked that we have just three states and just one _in-progress_, so we do not have to introduce a constant list of statuses yet.